pilot/pkg/status/reporter.go

```diff
@@ -342,3 +346,7 @@ func (r *Reporter) RegisterDisconnect(conID string, types []xds.EventType) {
 		delete(r.status, key)
 	}
 }
+
+func (r *Reporter) SetController(controller *DistributionController) {
+	r.controller = controller
```

Comment:
While this technically does read and write from the same value in different threads, the value is a pointer and the reads are all protected by a nil check, and the pointer is never set back to nil.  I can wrap it in a lock if you prefer, but it will have no effect on execution.

Reply:
is it a race to call SetController multiple times with the same value? since that can happen if we lose leader election.

On that note, what will happen when the controller stops but the reporter does not?

Reply:
if I understand right, stop will set state=nil which will then cause a nil pointer deref

Reply:
stopping the controller has no effect on state.  It simply stops the writeStatus loop.  No worries there.